files/image_config/reset-factory/reset-factory

```diff
@@ -0,0 +1,192 @@
+    fi
+    # Get default user accounts
+    default_users=$(jq -r '. | keys[]' $DEFAULT_USERS_FILE)
+    EXCLUDE_LIST=$(echo $default_users | tr ' ' '|')
+    # Get non-default user accounts
+    other_users=$(getent passwd | awk -F:  '($3>=1000 && $3<=6000) {print $1}' | grep -E -v $EXCLUDE_LIST)
+    echo "Remove non-default users"
+    for user in ${other_users[@]}
+    do
+       # avoid printing home directory and mail spool errors
+       userdel -rf $user 2> /dev/null
+    done
+    echo "Restore default users passwords"
+    for user in ${default_users[@]}
+    do
+       user_pass=$(jq -r '.[$user].password' --arg user "${user}" $DEFAULT_USERS_FILE)
+       echo "$user:$user_pass" | chpasswd -e
+    done
+}
+
+# Only root can run reset factory
+if [ $UID != 0 ]; then
+    echo "You must be root to reset system to factory settings"
+    exit 1
+fi
+
+CMD=$1
+FACTORY_TYPE=
+
+if [ "$CMD" = "keep-all-config" ] || [ "$CMD" = "only-config" ] || \
+   [ "$CMD" = "keep-basic" ] || [ -z "$CMD" ]; then
+    FACTORY_TYPE=$CMD
+else
+    usage
+    exit 1
+fi
+
+SERVICES_STOPPED=1
+echo "Stop critical services"
+monit unmonitor container_checker
+systemctl stop sonic.target --job-mode replace-irreversibly
+
+rc=$?
+if [ $rc -ne 0 ]; then
+    error_cleanup
+fi
+
+DATE=$(date "+%Y/%m/%d %H:%M:%S")
+HOSTNAME=$(hostname | sed 's/\./ /' | awk '{print $1}')
+printf "%s %s reset-factory: resetting system to factory defaults\n" "$DATE" "$HOSTNAME" >> $PERMLOG
+
+# Backup and delete config_db.json
+TEMP_CFG="/tmp/temp_config_db.$$"
+cp ${CONFIG_DB_JSON} ${TEMP_CFG}
+if [ "$FACTORY_TYPE" != "keep-basic" ] && [ "$FACTORY_TYPE" != "keep-all-config" ]; then
+    rm -f ${CONFIG_DB_JSON}
+fi
+
+echo "Call config-setup factory"
+config-setup factory $FACTORY_TYPE
+rc=$?
+if [ $rc -ne 0 ]; then
+    error_cleanup
+fi
+
+if [ "$FACTORY_TYPE" != "only-config" ]; then
+
+    if [ "$FACTORY_TYPE" != "keep-basic" ]; then
+
+        # Delete non-default users and restore default users passwords
+        reset_users
+
+        echo "Delete bash, python and vim history files"
+        find /home /root -type f -name ".bash_history" -o -name ".python_history" -o -name ".viminfo" | xargs rm -rf
+
+        echo "Delete any non-dotfiles in users home directories"
+        find /home/ /root -type f ! -iname ".*" -delete
+    fi
+
+    echo "Remove all docker containers"
+    docker rm -f $(docker ps -a -q | grep -v $(docker ps -a -q -f "name=database")) > /dev/null
+
+    echo "Clear sonic directory"
+    clear_sonic_dir
+
+    echo "Clear warmboot folder"
+    find /host/warmboot/ -type f -delete
+
+    echo "Delete reboot-cause files and symlinks"
+    find /host/reboot-cause/ -type l,f -delete
+
+    echo "Delete tech-support files"
+    rm -rf /var/dump/*
+
+    echo "Delete logs files"
+    find /var/log/ -type f ! -iname "wtmp" ! -iname "btmp" ! -iname "lastlog" ! -iname "systemlog" -delete
+
+    # Clear wtmp, utmp and lastlog files
+    rm -rf /var/log/btmp.*
```

Comment:
Why do we need special handling for these files? Any side effects of deleting them altogether?

Reply:
Yes I saw in few places that it is not recommended to delete these files.
For example: [btmp](https://askubuntu.com/questions/620458/is-it-safe-to-delete-btmp-from-log-on-ubuntu-14-04-rackapace-server)
So first, we delete the rotation files ( in case user configured rotation rule for these files)
and then empty them